dbms/src/Flash/Mpp/MinTSOScheduler.cpp

```diff
@@ -178,51 +176,64 @@ void MinTSOScheduler::deleteQuery(
 }
 
 /// NOTE: should not throw exceptions due to being called when destruction.
-void MinTSOScheduler::releaseThreadsThenSchedule(const int needed_threads, MPPTaskManager & task_manager)
+void MinTSOScheduler::releaseThreadsThenSchedule(
+    const String & resource_group_name,
+    int needed_threads,
+    MPPTaskManager & task_manager)
 {
     if (isDisabled())
     {
         return;
     }
 
-    auto updated_estimated_threads = static_cast<Int64>(estimated_thread_usage) - needed_threads;
+    global_estimated_thread_usage -= needed_threads;
+    auto & entry = getOrCreateGroupEntry(resource_group_name);
+    auto updated_estimated_threads = static_cast<Int64>(entry.estimated_thread_usage) - needed_threads;
     RUNTIME_ASSERT(
```

Comment:
It seems if the group entry is created in L190, the assert must fail?

Reply:
Why is this situation happening? Under normal circumstances, tasks are scheduled(group entry create and estimated_thread_usage added) first before releasing threads

Reply:
add `mustGetGroupEntry`